internal/pkg/aws/codepipeline/codepipeline.go

```diff
@@ -191,3 +181,52 @@ func (c *CodePipeline) getPipelineName(resourceArn string) (string, error) {
 
 	return parsedArn.Resource, nil
 }
+
+// GetPipelineStatus retrieves status information from a given pipeline.
+func (c *CodePipeline) GetPipelineState(name string) (*PipelineState, error) {
+	input := &cp.GetPipelineStateInput{
+		Name: aws.String(name),
+	}
+	resp, err := c.client.GetPipelineState(input)
+
+	if err != nil {
+		return nil, fmt.Errorf("get pipeline state %s: %w", name, err)
+	}
+	var stageStates []*StageState
+	for _, stage := range resp.StageStates {
+		var bool string
+		if *stage.InboundTransitionState.Enabled == true {
+			bool = "ENABLED"
+		} else {
+			bool = "DISABLED"
+		}
+		var status string
+		for _, actionState := range stage.ActionStates {
+			if actionState.LatestExecution == nil {
+				status = "  -"
+			} else {
+				status = aws.StringValue(actionState.LatestExecution.Status)
+			}
+		}
+
+		stageToAdd := &StageState{
+			StageName:  aws.StringValue(stage.StageName),
+			Status:     status,
+			Transition: bool,
+		}
+		stageStates = append(stageStates, stageToAdd)
+	}
+	pipelineState := &PipelineState{
+		PipelineName: aws.StringValue(resp.PipelineName),
+		StageStates:  stageStates,
+		UpdatedAt:    resp.Updated,
+	}
+	return pipelineState, nil
+}
+
+// HumanString returns the stringified PipelineState struct with human readable format.
+// Example output:
+//   DeployTo-test	Deploy	Cloudformation	stackname: dinder-test-test
+func (s *StageState) HumanString() string {
+	return fmt.Sprintf("  %s\t%s\t%s\n", s.StageName, s.Status, s.Transition)
```

Comment:
Is there a reason for the space in front?

Reply:
It indents the output so it's lined up with the column headings.